nemo/collections/asr/parts/submodules/rnnt_beam_decoding.py

```diff
@@ -1196,7 +1206,12 @@ def modified_adaptive_expansion_search(
                                     lm_score, new_hyp.ngram_lm_state = self.compute_ngram_score(
                                         hyp.ngram_lm_state, int(k)
                                     )
-                                    new_hyp.score += self.ngram_lm_alpha * lm_score
+                                    if isinstance(self.joint, hat.HATJoint) and self.hat_subtract_ilm:
```

Comment:
Same for everywhere else below.

Reply:
Hi @titu1994! Thank you for detailed review. I tried to modify HAT related code according to your suggestions. For convenience I also added `resolve_joint_output` function. Check it pls.